sdk/tables/azure-data-tables/azure/data/tables/_encoder.py

```diff
@@ -168,18 +173,15 @@ def encode_entity(self, entity: Union[TableEntity, Mapping[str, Any]]) -> Dict[s
         for key, value in entity.items():
             edm_type, value = self.prepare_value(key, value)
             try:
-                if _ODATA_SUFFIX in key or key + _ODATA_SUFFIX in entity:
+                if _ODATA_SUFFIX in key or f"{key}{_ODATA_SUFFIX}" in entity:
                     encoded[key] = value
                     continue
                 # The edm type is decided by value
                 # For example, when value=EntityProperty(str(uuid.uuid4), "Edm.Guid"),
                 # the type is string instead of Guid after encoded
                 if edm_type:
-                    encoded[key + _ODATA_SUFFIX] = edm_type.value if hasattr(edm_type, "value") else edm_type
+                    encoded[f"{key}{_ODATA_SUFFIX}"] = edm_type.value if hasattr(edm_type, "value") else edm_type
```

Comment:
If you use the same f-string twice, keep it in a variable, so it reduces the risk of type and mismatch later if we have to touch this code.

Reply:
Updated